### ISSUE TITLE
About the evaluation of random，centroids and neighbors

### ISSUE THREAD

**Issue by xqx456** [GeorgeCazenavette/linear-gradient-matching#3]
Hello, LGM is a very valuable and practical work, and I'm very grateful for sharing such clear code. I have a question for you.

I've repeatedly tried using random images instead of tensors for training in eval.py, and the performance has consistently been below 2%. I've changed the image loading code as follows: 

transform = transforms.Compose([
        transforms.Resize((cfg.real_res, cfg.real_res)), 
        (
                    transforms.CenterCrop(cfg.crop_res)
                    if cfg.train_crop_mode == "center"
                    else transforms.RandomCrop(cfg.crop_res)
                ),
        transforms.ToTensor(),         
        transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]) 
    ])

    ds = datasets.ImageFolder(root=cfg.syn_path, transform=transform)
    images = torch.stack([img for img, _ in ds]).cuda()
    labels = torch.tensor([label for _, label in ds]).cuda()

    ds_gpu = TensorDataset(images.detach().clone(), labels.detach().clone())

    loader = DataLoader(ds_gpu, batch_size=min(100, len(ds_gpu)), shuffle=True)

Could you please explain how you preprocess the randomly selected or center-selected images, and what might be causing the performance issues?

**Comment by GeorgeCazenavette** [comment]
Hi! So sorry I missed this issue.

I'm a little confused on what exactly you're doing. Are you saying that you're trying to load random real images, train a linear head on them, and then evaluate?

If so, we have a script for that at `baselines/random_reals.py`. 

We also have instructions on how to run the other baselines [here](https://github.com/GeorgeCazenavette/linear-gradient-matching/tree/main?tab=readme-ov-file#baselines) in the README. 

Does that help with your question?

**Comment by fglore6-prog** [comment]
> Hello, LGM is a very valuable and practical work, and I'm very grateful for sharing such clear code. I have a question for you.
> 
> I've repeatedly tried using random images instead of tensors for training in eval.py, and the performance has consistently been below 2%. I've changed the image loading code as follows: 
> 
> transform = transforms.Compose([
>         transforms.Resize((cfg.real_res, cfg.real_res)), 
>         (
>                     transforms.CenterCrop(cfg.crop_res)
>                     if cfg.train_crop_mode == "center"
>                     else transforms.RandomCrop(cfg.crop_res)
>                 ),
>         transforms.ToTensor(),         
>         transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]) 
>     ])
> 
>     ds = datasets.ImageFolder(root=cfg.syn_path, transform=transform)
>     images = torch.stack([img for img, _ in ds]).cuda()
>     labels = torch.tensor([label for _, label in ds]).cuda()
> 
>     ds_gpu = TensorDataset(images.detach().clone(), labels.detach().clone())
> 
>     loader = DataLoader(ds_gpu, batch_size=min(100, len(ds_gpu)), shuffle=True)
> 
> Could you please explain how you preprocess the randomly selected or center-selected images, and what might be causing the performance issues?

**Comment by fglore6-prog** [comment]
> Hello, LGM is a very valuable and practical work, and I'm very grateful for sharing such clear code. I have a question for you.
> 
> I've repeatedly tried using random images instead of tensors for training in eval.py, and the performance has consistently been below 2%. I've changed the image loading code as follows: 
> 
> transform = transforms.Compose([
>         transforms.Resize((cfg.real_res, cfg.real_res)), 
>         (
>                     transforms.CenterCrop(cfg.crop_res)
>                     if cfg.train_crop_mode == "center"
>                     else transforms.RandomCrop(cfg.crop_res)
>                 ),
>         transforms.ToTensor(),         
>         transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]) 
>     ])
> 
>     ds = datasets.ImageFolder(root=cfg.syn_path, transform=transform)
>     images = torch.stack([img for img, _ in ds]).cuda()
>     labels = torch.tensor([label for _, label in ds]).cuda()
> 
>     ds_gpu = TensorDataset(images.detach().clone(), labels.detach().clone())
> 
>     loader = DataLoader(ds_gpu, batch_size=min(100, len(ds_gpu)), shuffle=True)
> 
> Could you please explain how you preprocess the randomly selected or center-selected images, and what might be causing the performance issues?

**Comment by fglore6-prog** [comment]
- > []()
